### PR TITLE
Simplify weaviate auth

### LINKED ISSUE #222: Feature Request: Simplify weaviate auth
To help new users get started with weaviate, we should simplify how they set up the authentication.

In particular, we should only support the following ways to connect to weaviate:

* WCS via API key only

* Docker compose unauthenticated only

This means users only need the following environment variables:

* `WEAVIATE_URL`: can be e.g.localhost:8080 or the WCS url

* `WEAVIATE_API_KEY`: the API Key if using WCS

### PR DESCRIPTION
**Pull request by hsm207**
Fixes #222

### CODE REVIEW

File: datastore/providers/weaviate_datastore.py [hunk 1]
```diff
@@ -370,3 +369,17 @@ def _is_valid_weaviate_id(candidate_id: str) -> bool:
                 return True
         except ValueError:
             return False
+
+    @staticmethod
+    def _is_wcs_domain(url: str) -> bool:
```
Comment (isafulf) [inline datastore/providers/weaviate_datastore.py:374]: what if it is a WCS enterprise cluster? This regex won't work

**Review by isafulf — COMMENTED** [review]
LGTM! Thanks for simplifying

File: datastore/providers/weaviate_datastore.py (hunk 1 above)
Comment (byronvoorbach) [reply datastore/providers/weaviate_datastore.py:374]: Good find! Let me get back to you on this

Comment (hsm207) [reply datastore/providers/weaviate_datastore.py:374]: done!

**Pull request merged**